nautobot_ssot/models.py

```diff
@@ -33,6 +40,116 @@
 from .choices import SyncLogEntryActionChoices, SyncLogEntryStatusChoices
 
 
+class CompressedBinaryField(models.BinaryField):
+    """
+    Compress a binary field.
+
+    This class is adapted from https://gist.github.com/tomfa/665f8a655a9218e0b4e9bd394d459934
+    under the MIT License (https://gist.github.com/tomfa).
+    """
+
+    compress = compress_string
+
+    @staticmethod
+    def uncompress(compressed_value: Union[bytes, bytearray, memoryview]) -> str:
+        """
+        Uncompress stored value for compessed field.
+
+        Args:
+            compressed_value: The compressed data stored in the database.
+
+        Returns:
+            str: The `compressed_value` uncompressed.
+        """
+        zbuf = io.BytesIO(compressed_value)
+        zfile = gzip.GzipFile(fileobj=zbuf)
+        try:
+            uncompressed_value = zfile.read()
+        finally:
+            zfile.close()
```

Comment:
Can it be used as a context manager?

```suggestion
        with gzip.GzipFile(fileobj=zbuf) as zfile:
            uncompressed_value = zfile.read()
```

Reply:
I looked into this, and the way that we have to use it does not support a context manager.